kafka/kafka.go

```diff
@@ -276,11 +276,12 @@ const PartitionAny = int32(C.RD_KAFKA_PARTITION_UA)
 
 // TopicPartition is a generic placeholder for a Topic+Partition and optionally Offset.
 type TopicPartition struct {
-	Topic     *string
-	Partition int32
-	Offset    Offset
-	Metadata  *string
-	Error     error
+	Topic       *string
+	Partition   int32
+	Offset      Offset
+	Metadata    *string
+	Error       error
+	LeaderEpoch *int32
```

Comment:
Can this just be int32, we can use a negative value (-1) similar to C case (as illustrated in`if cLeaderEpoch >= 0`).
We can change if it makes sense.